app/javascript/utils/lazy-highlight-all.ts

```diff
@@ -0,0 +1,54 @@
+export function lazyHighlightAll(): void {
+  let highlighted = false
+
+  function loadAndHighlight() {
+    import('@/utils/highlight')
+      .then((m) => {
+        m.highlightAll()
+        observer.disconnect()
+      })
+      .catch((e) => {
+        observer.disconnect()
+        throw new Error(e)
+      })
+  }
+
+  function checkAndHighlight(): void {
+    // only load highlightjs when a code block with lang-... or language-... classname exists in the DOM
+    if (highlighted || !hasHighlightableCodeBlock(document)) return
+
+    // to avoid race conditions, set this optimistically to true before the async loading
+    highlighted = true
+    loadAndHighlight()
+  }
+
+  function mutationCallback(mutationsList: MutationRecord[]): void {
+    for (const mutation of mutationsList) {
+      if (mutation.type !== 'childList') continue
+      for (const node of mutation.addedNodes) {
+        if (node.nodeType !== Node.ELEMENT_NODE) continue
+        if (hasHighlightableCodeBlock(node as Element)) {
+          checkAndHighlight()
+          return
+        }
+      }
+    }
+  }
+
+  checkAndHighlight()
+
+  const observer = new MutationObserver(mutationCallback)
+  observer.observe(document.body, { childList: true, subtree: true })
+
+  // if no code block is found within 2 seconds after the turbo:load event, disconnect the observer.
+  setTimeout(() => {
+    observer.disconnect()
+  }, 2000)
+}
+
+function hasHighlightableCodeBlock(rootNode: Document | Element): boolean {
+  return (
+    rootNode.querySelector('code[class^=lang-]') !== null ||
+    rootNode.querySelector('code[class^=language-]') !== null
+  )
```

Comment:
Can some codeblocks not have languages attached? Do we normally rely on highlightjs to auto-determine the language in these situations? Should we just check for `code`?

Reply:
> Should we just check for `code`

That's a good suggestion.

Reply:
@dem4ron I don't think there's any real downside in just simplifying this to `code`. It removes the risk of edge-cases we've not thought of.

Reply:
okay!